python/link_prediction.py

```diff
@@ -0,0 +1,771 @@
+        optimizer: str -> Can be one of the following: ADAM, SGD, AdaGrad...
+        learning_rate: float -> Learning rate for optimizer
+        split_ratio: float -> Split ratio between training and validation set. There is not test dataset because it is assumed that user first needs to create new edges in dataset to test a model on them.
+        node_features_property: str → Property name where the node features are saved.
+        device_type: str ->  If model will be trained using CPU or cuda GPU. Possible values are cpu and cuda. To run it on Cuda, user must set this flag to true and system must support cuda execution.
+                                System's support is checked with torch.cuda.is_available()
+        console_log_freq: int ->  how often do you want to print results from your model? Results will be from validation dataset.
+        checkpoint_freq: int → Select the number of epochs on which the model will be saved. The model is persisted on disc.
+        aggregator: str → Aggregator used in models. Can be one of the following: lstm, pool, mean, gcn.
+        metrics: mgp.List[str] -> Metrics used to evaluate model in training.
+        predictor_type str: Type of the predictor. Predictor is used for combining node scores to edge scores.
+        attn_num_heads: List[int] -> GAT can support usage of more than one head in each layer except last one. Only used in GAT, not in GraphSage.
+        tr_acc_patience: int -> Training patience, for how many epoch will accuracy drop on test set be tolerated before stopping the training.
+        model_save_path: str -> Path where the link prediction model will be saved every checkpoint_freq epochs.
+
+
+    Returns:
+        mgp.Record:
+            status (bool): True if parameters were successfully updated, False otherwise.
+            message(str): Additional explanation why method failed or OK otherwise.
+    """
+    global link_prediction_parameters
+
+    print("START")
+    print(link_prediction_parameters)
+
+    validation_status, validation_message = _validate_user_parameters(parameters=parameters)
+    if validation_status is False:
+        return mgp.Record(status=validation_status, message=validation_message)
+
+    for key, value in parameters.items():
+        if not hasattr(link_prediction_parameters, key):
+            return mgp.Record(
+                status=1,
+                message="No attribute " + key + " in class LinkPredictionParameters",
+            )
+        try:
+            setattr(link_prediction_parameters, key, value)
+        except Exception as exception:
+            return mgp.Record(status=1, message=repr(exception))
+
+    # Device type handling
+    if link_prediction_parameters.device_type == CUDA_DEVICE and torch.cuda.is_available() is True:
+        link_prediction_parameters.device_type = CUDA_DEVICE
+    else:
+        link_prediction_parameters.device_type = CPU_DEVICE
```

Comment:
Currently, GPU is not supported at all since I don't have cuda avinlable in my system and it is quite a burden to set up everything at Colab. But once the support is added, the predict method should also be GPU compatible.